Meta/CMake/common_compile_options.cmake

```diff
@@ -5,42 +5,54 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_COLOR_DIAGNOSTICS ON)
 
-add_compile_options(-Wall)
-add_compile_options(-Wextra)
+# FIXME: when compiing using cl-clang, the code tries to compile c++98 mode, this generates too much warnings
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang$" AND NOT CMAKE_CXX_SIMULATE_ID MATCHES "MSVC")
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+endif()
```

Comment:
This is not right.

- Building with the incorrect C++ standard can have consequences other than *warnings* being emitted for the use of newer features: which overloads are selected when multiple functions are viable, whether a `constexpr` function can actually be evaluated at compile time, etc. Yes, these are subtle issues, but we don't want to add a *different C++ standard* into our support matrix. Please try to debug why it's trying to build in C++98 mode (happy to chat on Discord about this).
- The way this condition is written means that most warnings will completely be disabled when building with GCC.

Reply:
Do not disagree. However, so far, I have no idea why `clang-cl` does not get that I am requesting C++23. Its on the top of toplevel CMake, and also on lagom. Happy for any input on that area.

Reply:
If someone else is reading this: we *were* actually setting the language standard correctly, it's just that in MSVC emulation mode, `-Wall` is equivalent to the unreasonably verbose `-Weverything`. According to the Clang docs, `/W4` should be what we need.

Also using clang.exe instead of clang-cl might be a good way to avoid these kinds of problems.

Reply:
yes, but clang uses gcc's ABI, and then we are limited to using 3rd party libraries built using GCC and not CL's ABI.

I *ideally* would like to have mingw builds of Ladybird... but, clang-cl seems like the best compromise.

Reply:
Clang uses the msvc abi if you pass it an msvc triple such as --target=x86_64-pc-windows-msvc (the "-msvc" at the end is even the default and you can omit it).

The `clang` driver does not have flags for everything that `clang-cl` has flags for, but probably enough to get something working. (And upstream will gladly accept patches for missing win abi driver flags for `clang`.) The main drawback is that `clang-cl` targeting win/Ms abi is heavily used and known to work well, while `clang` targeting the same is used much less and will have more unknown unknowns.